pkg/kubelet/kubelet.go

```diff
@@ -3020,12 +3024,29 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod, podStatus *kubecontaine
 	}
 
 	if reason != "" {
-		kl.statusManager.SetPodResizePendingCondition(pod.UID, reason, message)
+		kl.statusManager.SetPodResizePendingCondition(pod.UID, reason, message, pod.Generation)
+
+		// The new pod resize is either deferred or infeasible, which means if there is a pod
+		// resize in progress, it is associated with an older resize. In this case, the
+		// condition reflecting the older resize should also have the older observedGeneration.
+		generationForPodResizeInProgressCondition = kl.oldResizeInProgressGeneration(pod)
 	}
 
 	return podFromAllocation, nil
 }
 
+// oldResizeInProgressGeneration returns the ObservedGeneration of the existing PodResizeInProgress
+// condition if it exists. If the condition is not found, it returns the pod metadata.generation.
+func (kl *Kubelet) oldResizeInProgressGeneration(pod *v1.Pod) int64 {
+	conditions := kl.statusManager.GetPodResizeConditions(pod.UID)
```

Comment:
I don't think we're backfilling this on Kubelet restart, right? If so, the old conditions will be lost on Kubelet restart.

Reply:
....feel free to let me know if that's not the right place for it